src/components/Widgets/NumberControl.js

```diff
@@ -1,17 +1,34 @@
 import React, { PropTypes } from 'react';
 
-export default class StringControl extends React.Component {
+export default class NumberControl extends React.Component {
   handleChange = (e) => {
-    this.props.onChange(e.target.value);
+    const valueType = this.props.field.get('valueType')
```

Comment:
We still use semicolons everywhere, so we'll need them here, too. We'll probably consider dropping them at some point, I've certainly considered it.